components/brave_wallet_ui/common/hooks/swap.ts

```diff
@@ -584,45 +741,60 @@ export default function useSwap ({ fromAsset: fromAssetProp, toAsset: toAssetPro
   }, [orderType])
 
   const onSubmitSwap = React.useCallback(() => {
-    if (!swapQuote) {
-      return
-    }
-
     if (!fromAsset) {
       return
     }
 
-    if (swapValidationError === 'insufficientAllowance' && allowance) {
-      // IMPORTANT SECURITY NOTICE
-      //
-      // The token allowance suggested by Swap is always unlimited,
-      // i.e., max(uint256). While unlimited approvals are not safe from a
-      // security standpoint, and this puts the entire token balance at risk
-      // if 0x contracts are ever exploited, we still opted for this to give
-      // users a frictionless UX and save on gas fees.
-      //
-      // The transaction confirmation screen for ERC20 approve() shows a loud
-      // security notice, and still allows users to edit the default approval
-      // amount.
-      const allowanceHex = new Amount(MAX_UINT256)
-        .toHex()
-
-      dispatch(WalletActions.approveERC20Allowance({
-        from: selectedAccount.address,
-        contractAddress: fromAsset.contractAddress,
-        spenderAddress: swapQuote.allowanceTarget,
-        allowance: allowanceHex
-      }))
+    if (swapProvider === SwapProvider.ZeroEx) {
+      if (!swapQuote) {
+        return
+      }
 
+      if (swapValidationError === 'insufficientAllowance' && allowance) {
+        // IMPORTANT SECURITY NOTICE
+        //
+        // The token allowance suggested by Swap is always unlimited,
+        // i.e., max(uint256). While unlimited approvals are not safe from a
+        // security standpoint, and this puts the entire token balance at risk
+        // if 0x contracts are ever exploited, we still opted for this to give
+        // users a frictionless UX and save on gas fees.
+        //
+        // The transaction confirmation screen for ERC20 approve() shows a loud
```

Comment:
do you have a screenshot of this? cc @kdenhartog

Reply:
oh is it the screenshot in https://github.com/brave/security/issues/960 ? if so NVM

Reply:
Will give this a look to understand the tradeoff here a bit more and let you know what I think. Thanks for tagging me

Reply:
<img width="320" alt="Screenshot 2022-08-04 at 20 42 12" src="https://user-images.githubusercontent.com/3684187/182882847-3c152df9-b13d-48a5-8e24-cb926305642c.png">